src/Finder/CollectionFinder.php

```diff
@@ -40,7 +40,8 @@ private function hydrate($results)
         }
         $rsm = new ResultSetMappingBuilder($this->em);
         $rsm->addRootEntityFromClassMetadata($this->collectionConfig['entity'], 'e');
-        $query = $this->em->createNativeQuery('SELECT * FROM client WHERE id IN ('.implode(', ', $ids).') ORDER BY FIELD(id,'.implode(', ', $ids).')', $rsm);
+        $tableName = $this->em->getClassMetadata($this->collectionConfig['entity'])->getTableName();
+        $query = $this->em->createNativeQuery('SELECT * FROM '.$tableName.' WHERE id IN ('.implode(', ', $ids).') ORDER BY FIELD(id,'.implode(', ', $ids).')', $rsm);
```

Comment:
Do you always use ids to retrieve the entities or it's possible to use another field ? I'm not sure

Reply:
For now, it's always the ids :)

Reply:
Ok, so it should work, just a little issue, we have to handle when the array of ids is empty

Reply:
Hello, I made a mistake in my previous answer : its not always the id that is used to query, it the `$primaryKeyInfos['entityAttribute']` field

Reply:
Ok, so there is another issue, in the typesense bundle config I created this field in one of my collection : 
```
id: # Entity attribute name
          name: entity_id # Typesense attribute name
          type: primary # Attribute type
```
like this I can use it as default_sorting_field

but  when it's time to remove or update, the code is looking for a field id, so it was failing. In typesense the field "id" is generated automatically by typesense server because we didn't give it one. 

In typesense the default_sorting_field have to be an int32 or float and the id have to be a string. 

So I kept my entity_id field like this, and during the transform of doctrine entity to typesense document I'm adding a string field id which is the doctrine id. So, It's working, but I think we have to find a better way to do it.. Mostly if we want to handle primary key which are not called "id".

I hope I was clear :sweat_smile: